src/msg/async/AsyncMessenger.cc

```diff
@@ -357,6 +400,72 @@ int AsyncMessenger::shutdown()
   return 0;
 }
 
+void AsyncMessenger::dump(Formatter* f) {
+  std::lock_guard l{lock};
```

Comment:
For unstable situations, I'd be concerned that the output of this dump may be significant. Perhaps a `flags` argument to select which arrays below to expand/dump?

Reply:
True. The connections map alone may have 1000s of entries on larger clusters. The latest changes add filters/flags for most larger arrays and a flag for including TCP INFO.